frontend/tests/components/ImageForm.test.jsx

```diff
@@ -0,0 +1,127 @@
+})
+
+test('image note can be written', async() => {
+  const createImage = vi.fn()
+  const user = userEvent.setup()
+
+  render(<ImageForm createImage={createImage}/>)
+
+  const noteInput = screen.getByLabelText('Note:')
+
+  await user.type(noteInput, "this is a note")
+
+  expect(noteInput.value).toBe("this is a note")
+})
+
+test('note is required', async() => {
+  const createImage = vi.fn()
+  const user = userEvent.setup()
+
+  render(<ImageForm createImage={createImage}/>)
+
+  const submit = screen.getByText('Save')
+  const imageSelector = screen.getByLabelText('Select image:')
+  const file = new File(['hello'], 'hello.png', {type: 'image/png'})
+
+  await user.upload(imageSelector, file)
+  await user.click(submit)
+
+  expect(imageSelector.files[0]).toBe(file)
+  expect(imageSelector.files.length).toBe(1)
+})
+
+test('image is required', async() => {
+  const createImage = vi.fn()
+  const user = userEvent.setup()
+
+  render(<ImageForm createImage={createImage}/>)
+
+  const submit = screen.getByText('Save')
+  const noteInput = screen.getByLabelText('Note:')
+  const imageSelector = screen.getByLabelText('Select image:')
+
+  await user.type(noteInput, "this is a note")
+  await user.click(submit)
+
+  expect(noteInput.value).toBe("this is a note")
+  expect(imageSelector.files.length).toBe(0)
+})
+
+// During this test the imageSelector holds the file as expected
+// but when checked in submit-phase it does not exist.
+/* 
+  test('image and note can be submitted', async() => {
+  const createImage = vi.fn()
+  const user = userEvent.setup()
+
+  render(<ImageForm createImage={createImage}/>)
+
+  const imageSelector = screen.getByLabelText('Select image:')
+  const noteInput = screen.getByLabelText('Note:')
+  const submit = screen.getByText('Save')
+
+  const file = new File(['hello'], 'hello.png', {type: 'image/png'})
+
+  await waitFor(() => user.upload(imageSelector, file))
+  expect(imageSelector.files[0]).toBe(file)
+  console.log("filename:",imageSelector.files[0].name)
+  console.log("missing:", imageSelector.validity.valueMissing)
+  console.log("selector:", imageSelector.validationMessage)
+  await user.type(noteInput, "this is a note")
+  await user.click(submit)
+
+  expect(createImage.mock.calls).toHaveLength(1)
+  expect(createImage.mock.calls[0][0]).toEqual({note: "this is a note", image: file})
+}) 
+*/
+
+/*
+test("non-images can't be submitted", async() => {
+  const createImage = vi.fn()
+  const user = userEvent.setup()
+
+  render(<ImageForm createImage={createImage}/>)
+
+  const imageSelector = screen.getByLabelText('Select image:')
+  const noteInput = screen.getByLabelText('Note:')
+  const submit = screen.getByText('Save')
+
+  const file = new File(['hello'], 'hello.txt', {type: 'text/plain'})
+
+  await user.upload(imageSelector, file)
+  await user.type(noteInput, "this is a note")
+  await user.click(submit)
+
+  expect(createImage.mock.calls).toHaveLength(0)
+})
+*/
```

Comment:
To keep the test file clean, could we remove these commented-out tests? Perhaps they could be turned into a separate issue?

Reply:
Done